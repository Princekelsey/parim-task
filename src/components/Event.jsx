import React from "react";
import moment from "moment";
import { FaCalendarAlt } from "react-icons/fa";

const Event = ({ date, event }) => {
  const day = moment(date).format("dddd");
  return (
    <div className="col-1-of-4 mb-sm">
      <div className="feature-box">
        <FaCalendarAlt className="feature-box__icon" />
        <h2>{day}</h2>
        <h3 className="heading-tertiary mb-md"> {date}</h3>
        {event.length ? (
          <>
            <h4>Event(s)</h4>
            {event.map((e, index) => (
              <ul key={index}>
                <li className="mb-sm">
                  <strong>Type: </strong>
                  {e.type} <br></br>
                  <strong>Name: </strong>
                  {e.name}
                </li>
              </ul>
            ))}{" "}
          </>
        ) : (
          <p className="feature-box__text">No event(s) on this date</p>
        )}
      </div>
    </div>
  );
};

export default Event;
