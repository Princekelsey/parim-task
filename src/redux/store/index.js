import { createStore, applyMiddleware } from "redux";
import logger from "redux-logger";
import createSagaMiddleware from "redux-saga";
import rootReducer from "../rootReducer";
import rootSaga from "../rootSaga";

// initailize saga middleware
const sagaMiddleware = createSagaMiddleware();

// all middlewares
const middleWares = [sagaMiddleware];

// check if development mode and use logger
if (process.env.NODE_ENV === "development") {
  middleWares.push(logger);
}

// redux store
const store = createStore(rootReducer, applyMiddleware(...middleWares));

// run saga middleware
sagaMiddleware.run(rootSaga);

export default store;
