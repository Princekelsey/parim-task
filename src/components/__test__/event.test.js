import React from "react";
import ReactDOM from "react-dom";
import Event from "../Event";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../redux/store";

afterEach(cleanup);
const mockEvent = [{ type: "test", name: "testName" }];

it("Event component renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Provider store={store}>
      <Event date="2020-06-09" event={mockEvent} />
    </Provider>,
    div
  );
});

it("renders calender correctly", () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <Event date="2020-06-09" event={mockEvent} />
    </Provider>
  );
  expect(getByTestId("event"));
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Event date="2020-06-09" event={mockEvent} />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
