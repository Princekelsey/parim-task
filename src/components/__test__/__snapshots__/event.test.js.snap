// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="col-1-of-4 mb-sm"
  data-testid="event"
>
  <div
    className="feature-box"
  >
    <svg
      className="feature-box__icon"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm320-196c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM192 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12h-40c-6.6 0-12-5.4-12-12v-40zM64 268c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zm0 128c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H76c-6.6 0-12-5.4-12-12v-40zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
      />
    </svg>
    <h2>
      Tuesday
    </h2>
    <h3
      className="heading-tertiary mb-md"
    >
       
      2020-06-09
    </h3>
    <h4>
      Event(s)
    </h4>
    <ul>
      <li
        className="mb-sm"
      >
        <strong>
          Type: 
        </strong>
        test
         
        <br />
        <strong>
          Name: 
        </strong>
        testName
      </li>
    </ul>
     
  </div>
</div>
`;
